wallet/udb/txmined.go

```diff
@@ -2816,24 +2822,110 @@ func (s *Store) MakeInputSource(ns, addrmgrNs walletdb.ReadBucket, account uint3
 			var op wire.OutPoint
 			err = readCanonicalOutPoint(k, &op)
 			if err != nil {
-				return 0, nil, nil, err
+				return nil, err
 			}
 			op.Tree = tree
 
 			input := wire.NewTxIn(&op, int64(amt), nil)
 
+			// Unspent credits are currently expected to be P2PKH only.
+			// Multisig outputs have a different criteria to be classified as
+			// credits and stake related P2SH outputs are only spendable at
+			// specific points in the staking process.
+			scriptClass := txscript.GetScriptClass(txscript.DefaultScriptVersion, pkScript)
+			if scriptClass != txscript.PubKeyHashTy {
+				return nil, fmt.Errorf("unexpected script class encountered for credit: %v",
+					scriptClass.String())
+			}
+
 			currentTotal += amt
 			currentInputs = append(currentInputs, input)
 			currentScripts = append(currentScripts, pkScript)
+			redeemScriptSizes = append(redeemScriptSizes, txsizes.RedeemP2PKHSigScriptSize)
+		}
+
+		// Return the current results if the target was specified and met.
+		if target != 0 && currentTotal >= target {
+			inputDetail = txauthor.InputDetail{
+				Amount:            currentTotal,
+				Inputs:            currentInputs,
+				Scripts:           currentScripts,
+				RedeemScriptSizes: redeemScriptSizes,
+			}
+
+			return &inputDetail, nil
+		}
+
+		// Iterate through multisig outputs.
```

Comment:
don't change the semantics of this function by including outputs it didn't previous return.  This is also unrelated to the txauthor API change.